micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -648,12 +652,26 @@ private Meter getOrCreateMeter(@Nullable DistributionStatisticConfig config,
                     }
                     meterMap.put(mappedId, m);
                 }
+                else {
+                    checkAndWarnAboutGaugeDoubleRegistration(m);
+                }
             }
         }
+        else {
+            checkAndWarnAboutGaugeDoubleRegistration(m);
```

Comment:
A nit-pick: this and the other added `else` are far from their corresponding `if`, making it hard to know the condition when this code path will be taken while reading the code. Since these are both one-liners, could we invert the `if` conditions, put these one-liners in it and make the existing `if` branches the `else` branches instead? I think that would be more readable.

Reply:
Good idea!